### PR TITLE
Fix issue with resource files not loaded on Ubuntu Linux

### PR DESCRIPTION
**Pull request by ghost**
On Ubuntu Linux the attempt to enumerate DAO block files in a JAR resource directory fails.

See https://github.com/bisq-network/bisq/pull/5894#issuecomment-986164443

All the block data is then downloaded from the seednodes.

This PR does the enumeration using `nio.File.FileSystem` so that block files in resources can be used.

### CODE REVIEW

File: core/src/main/java/bisq/core/dao/state/storage/BsqBlocksStorageService.java [hunk 1]
```diff
@@ -121,22 +128,21 @@ void copyFromResources(String postFix) {
...
                 return;
             }
-            File dir = new File(dirUrl.toURI());
-            String[] fileNames = dir.list();
-            if (fileNames == null) {
-                log.info("No files in directory. {}", dir.getAbsolutePath());
+            FileSystem fileSystem = FileSystems.newFileSystem(dirUrl.toURI(), Collections.emptyMap());
+            List<Path> filePaths = Files.walk(fileSystem.getPath(resourceDir), 1).collect(Collectors.toList());
+            if (filePaths.size() <= 1) {
+                log.info("No files in directory. {}", dirUrl.toString());
```
Comment (ripcurlx) [inline core/src/main/java/bisq/core/dao/state/storage/BsqBlocksStorageService.java:134]: toString is not necessary.

```suggestion
                log.info("No files in directory. {}", dirUrl);
```

**Comment by ripcurlx** [comment]
Doesn't work now on macOS unfortunately
```
java.lang.IllegalArgumentException: Path component should be '/'
	at java.base/sun.nio.fs.UnixFileSystemProvider.checkUri(UnixFileSystemProvider.java:81)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newFileSystem(UnixFileSystemProvider.java:90)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:338)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:287)
	at bisq.core.dao.state.storage.BsqBlocksStorageService.copyFromResources(BsqBlocksStorageService.java:131)
	at bisq.core.dao.state.storage.DaoStateStorageService.lambda$readFromResources$4(DaoStateStorageService.java:119)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

**Review by ripcurlx — CHANGES_REQUESTED** [review]
NACK - https://github.com/bisq-network/bisq/pull/5898#issuecomment-986659019

**Comment by stejbac** [comment]
I was debugging the same issue on Windows a couple of days ago, caused by the expression `new File(dirUrl.toUri())`. I ended up adapting the following method from StackOverflow, which seems to work:

https://stackoverflow.com/questions/3923129/get-a-list-of-resources-from-classpath-directory/48190582#48190582

It looks like you have to use two different code paths to list the resource directory, depending on the environment.

(I created a [draft patch for the fix](https://github.com/stejbac/bisq/commit/c8d0c49545bf0688e5789bec946102037c0151f6), but never got round to creating a PR out of it. The foreach loop to do the copying needs modifying to work on Windows as well, just like in this PR.)

**Comment by ghost** [comment]
@stejbac Looks like we were both debugging the same issue.  As this one does not work for MacOs, can you make your PR?

**Comment by stejbac** [comment]
@jmacxx OK, sure.

**Comment by chimp1984** [comment]
@stejbac Ah great that you tackled that as well. Could we extract the handling in a FileUtil class to make it re-usable. A pity that simple filehandling from resources is such a mess in java...
We have verious other loadFromResource calls. Maybe worth to re-check if there are issues as well.

**Pull request closed without merging**